extensions-core/google-extensions/src/main/java/org/apache/druid/data/input/google/GoogleCloudStorageEntity.java

```diff
@@ -33,36 +34,31 @@
 
 public class GoogleCloudStorageEntity extends RetryingInputEntity
 {
-  private final GoogleStorage storage;
-  private final URI uri;
+  private final GoogleByteSource byteSource;
 
-  GoogleCloudStorageEntity(GoogleStorage storage, URI uri)
+  GoogleCloudStorageEntity(GoogleStorage storage, CloudObjectLocation location)
   {
-    this.storage = storage;
-    this.uri = uri;
+    this.byteSource = new GoogleByteSource(storage, location.getBucket(), location.getPath());
   }
 
   @Nullable
   @Override
   public URI getUri()
   {
-    return uri;
+    return null;
```

Comment:
Doesn't look like the method is called in the codebase currently but this could construct a URI from the bucket/path if needed. 

`S3Entity` currently returns null for this as well, could probably have a comment on why in both locations

Reply:
I modified `GoogleCloudStorageEntity` and `S3Entity` to implement `getUri` using the `CloudObjectLocation.toUri` method